Testscripts/Linux/perf_apache.sh

```diff
@@ -0,0 +1,255 @@
+LogMsg "Configuring client ${client}..."
+install_apache
+if [ $? -ne 0 ]; then
+	LogMsg "Error: apache installation failed in ${client}.."
+	SetTestStateAborted
+	exit 1
+fi
+
+LogMsg "Configuring server ${server}..."
+Run_SSHCommand "${server}" ". $UTIL_FILE && install_apache"
+if [ $? -ne 0 ]; then
+	LogMsg "Error: apache installation failed in ${server}.."
+	SetTestStateAborted
+	exit 1
+fi
+
+GetDistro
+case "$DISTRO_NAME" in
+	oracle|rhel|centos)
+		web_server="httpd"
+		document_path="/var/www/html/test.dat"
+		;;
+
+	ubuntu|debian)
+		web_server="apache2"
+		document_path="/var/www/html/test.dat"
+		;;
+
+	sles|sle_hpc)
+		web_server="apache2"
+		document_path="/srv/www/htdocs/test.dat"
+		;;
+	*)
+		LogErr "Unsupported distribution for apache test"
+		SetTestStateSkipped
+		exit 1
+esac
+
+mpstat_cmd="mpstat"
+dstat_cmd="dstat"
+sar_cmd="sar"
+vmstat_cmd="vmstat"
+
+testName="apache"
+log_folder="$(pwd)/${testName}-test-logs"
+max_concurrency_per_ab=4
+max_ab_instances=16
+testConcurrency=(1 2 4 8 16 32 64 128 256 512 1024)
+
+Run_ab()
+{
+	current_concurrency=$1
+
+	if [ ${current_concurrency} -le 2 ]
+	then
+		total_requests=50000
+	elif [ ${current_concurrency} -le 128 ]
+	then
+		total_requests=100000
+	else
+		total_requests=200000
+	fi
+
+	ab_instances=$(($current_concurrency / $max_concurrency_per_ab))
+	if [ ${ab_instances} -eq 0 ]
+	then
+		ab_instances=1
+	fi
+	if [ ${ab_instances} -gt ${max_ab_instances} ]
+	then
+		ab_instances=${max_ab_instances}
+	fi
+
+	total_request_per_ab=$(($total_requests / $ab_instances))
+	concurrency_per_ab=$(($current_concurrency / $ab_instances))
+	concurrency_left=${current_concurrency}
+	requests_left=${total_requests}
+	while [ ${concurrency_left} -gt ${max_concurrency_per_ab} ]; do
+		concurrency_left=$(($concurrency_left - $concurrency_per_ab))
+		requests_left=$(($requests_left - $total_request_per_ab))
+		LogMsg "Running parallel ab command for: ${total_request_per_ab} X ${concurrency_per_ab}"
+		ab -n ${total_request_per_ab} -r -c ${concurrency_per_ab} http://${server}/test.dat >> ${log_folder}/${current_concurrency}.${testName}.bench.log & pid=$!
+		PID_LIST+=" $pid"
+	done
+
+	if [ ${concurrency_left} -gt 0 ]
+	then
+		LogMsg "Running parallel ab command left for: ${requests_left} X ${concurrency_left}"
+		ab -n ${requests_left} -r -c ${concurrency_left} http://${server}/test.dat >> ${log_folder}/${current_concurrency}.${testName}.bench.log & pid=$!
+		PID_LIST+=" $pid";
+	fi
+	trap "kill ${PID_LIST}" SIGINT
+	wait ${PID_LIST}
+}
+
+Run_apache()
+{
+	Run_SSHCommand "${server}" "pkill -f ab"
+	pkill -f ab
+
+	# Restart apache2 service
+	LogMsg "Info: Generate test data file on the Apache server /var/www/html/test.dat"
+	Run_SSHCommand "${server}" "dd if=/dev/urandom of=${document_path} bs=1K count=200"
+	Run_SSHCommand "${server}" "service ${web_server} stop"
+	Run_SSHCommand "${server}" "service ${web_server} start"
+	
+	Run_SSHCommand "${server}" "mkdir -p ${log_folder}"
```

Comment:
Is it not good to delete the folder to clean up and existing logs ?

Reply:
Fixed